pyxem/data/_data.py

```diff
@@ -0,0 +1,311 @@
+    ----------
+    **kwargs
+        Keyword arguments passed to :func:`~hyperspy.api.load`.
+    Examples
+    --------
+    >>> import pyxem as pxm
+    >>> s = pxm.data.pdnip_glass()
+    >>> s
+    >>> s.plot()
+    """
+    import zarr
+
+    pdnip = Dataset("PdNiP.zspy")
+    file_path = pdnip.fetch_file_path(allow_download=allow_download)
+    store = zarr.ZipStore(file_path)  # for loading from zip
+    return hs.load(store, **kwargs)
+
+
+def zrnb_percipitate(allow_download=False, **kwargs):
+    """A small 4-D STEM dataset of a ZrNb precipitate for strain mapping.
+
+    Parameters
+    ----------
+    **kwargs
+        Keyword arguments passed to :func:`~hyperspy.api.load`.
+    Examples
+    --------
+    >>> import pyxem as pxm
+    >>> s = pxm.data.zrnb_percipitate()
+    >>> s
+    >>> s.plot()
+    """
+    import zarr
+
+    pdnip = Dataset("PdNiP.zspy")
+    file_path = pdnip.fetch_file_path(allow_download=allow_download)
+    store = zarr.ZipStore(file_path)  # for loading from zip
+    return hs.load(store, **kwargs)
+
+
+def twinned_nanowire(allow_download=False, **kwargs):
+    """A small 4-D STEM dataset of a twinned nanowire for orientation mapping.
+
+    Parameters
+    ----------
+    **kwargs
+        Keyword arguments passed to :func:`~hyperspy.api.load`.
+    Examples
+    --------
+    >>> import pyxem as pxm
+    >>> s = pxm.data.twinned_nanowire()
+    >>> s
+    >>> s.plot()
+    """
+    nanowire = Dataset("twinned_nanowire.hdf5")
+    file_path = nanowire.fetch_file_path(allow_download=allow_download)
+    return hs.load(file_path, **kwargs)
+
+
+def sample_with_g(allow_download=False, **kwargs):
+    """A small 4-D STEM dataset of a twinned nanowire for orientation mapping.
+
+    Parameters
+    ----------
+    **kwargs
+        Keyword arguments passed to :func:`~hyperspy.api.load`.
+    Examples
+    --------
+    >>> import pyxem as pxm
+    >>> s = pxm.data.sample_with_g()
+    >>> s
+    >>> s.plot()
+    """
+    import zarr
+
+    sample = Dataset("sample_with_g.zspy")
+    file_path = sample.fetch_file_path(allow_download=allow_download)
+    store = zarr.ZipStore(file_path)  # for loading from zip
+    return hs.load(store, **kwargs)
+
+
+def mgo_nanocrystals(allow_download=False, **kwargs):
+    """A small 4-D STEM dataset of overlapping MgO nanocrystals
+
+    Parameters
+    ----------
+    **kwargs
+        Keyword arguments passed to :func:`~hyperspy.api.load`.
+    Examples
+    --------
+    >>> import pyxem as pxm
+    >>> s = pxm.data.mgo_nanocrystals()
+    >>> s
+    >>> s.plot()
+    """
+    import zarr
+
+    sample = Dataset("mgo_nano.zspy")
+    file_path = sample.fetch_file_path(allow_download=allow_download)
+    store = zarr.ZipStore(file_path)  # for loading from zip
+    return hs.load(store, **kwargs)
+
+
+class Dataset:
+    file_relpath: Path
+    file_package_path: Path
+    file_cache_path: Path
+    expected_md5_hash: str = ""
+    collection_name: Optional[str] = None
```

Comment:
No datasets are retreived from collections. I suggest to remove this part from the class to reduce complexity.

We use this in kikuchipy's data module when we want to download single files from a single Zenodo upload containing multiple files. For example when we want one EBSD dataset (.zip file) from a Zenodo upload containing ten datasets (as ten .zip files).